sdk/src/main/scala/com/horizen/SidechainNodeViewHolder.scala

```diff
@@ -103,95 +103,127 @@ class SidechainNodeViewHolder(sidechainSettings: SidechainSettings,
-              // since history and state are (according to the update procedure sequence: state --> wallet --> history)
-              // if necessary a rollback is applied internally to the forging box info storage, because
-              // it might have been updated upon consensus epoch switch even before the state
-              restoredWallet.ensureStorageConsistencyAfterRestore match {
-                case Success(checkedWallet) => {
-                  val checkedWalletVersion = checkedWallet.version
-                  log.info(s"walletVersion = ${checkedWalletVersion}")
-                  if (historyVersion == checkedWalletVersion) {
-                    // This is the successful case
-                    log.info("state, history and wallet storages are consistent")
-                    dumpStorages
-                    Some(restoredHistory, checkedState, checkedWallet, restoredMempool)
-                  }
-                  else {
-                    log.error("state and wallet storages are not consistent and could not be recovered")
-                    // wallet and state are not consistent, while state and history are, this should never happen
-                    // state --> wallet --> history
-                    None
-                  }
-                }
-                case Failure(e) => {
-                  log.error("wallet storages are not consistent", e)
-                  None
-                }
+
+        val historyStatus : Option[(String, Boolean, Int)] =  restoredHistory.getReindexStatus match {
+          case SidechainHistory.ReindexNotInProgress | 0 => {
+            Some(idToVersion(restoredHistory.bestBlockId), false, 0)
+          }
+          case reindexHeight => {
+            // best block id is updated in history storage as very last step
+            val blockIdOption = restoredHistory.getBlockIdByHeight(reindexHeight)
+            if (blockIdOption.isPresent){
+              Some(blockIdOption.get(), true, reindexHeight)
+            }else{
+              log.debug(s"Incosistent reindex state, unable to restore")
+              None
+            }
+          }
+        }
+        if (historyStatus.isEmpty){
+          None
+        }else {
+          val versionToAlign = historyStatus.get._1
+          val isReindexing = historyStatus.get._2
+          val reindexHeight = historyStatus.get._3
+
+          // get common version of the state storages, if necessary some rollback is applied internally
+          // according to the update procedure sequence
+          restoredState.ensureStorageConsistencyAfterRestore match {
+            case Success(checkedState) => {
+              val checkedStateVersion = checkedState.version
+
+              if (!isReindexing) {
+                log.debug(s"history bestBlockId = ${versionToAlign}, stateVersion = ${checkedStateVersion}")
+                val height_h = restoredHistory.blockInfoById(restoredHistory.bestBlockId).height
```

Comment:
height_h - why don't we use camelCase, such as heightHistory for example?

Reply:
old code coming from a previous ticket, I kept it unchanged